server/app/views/applicant/IneligibleBlockView.java

```diff
@@ -26,19 +30,62 @@ public final class IneligibleBlockView extends ApplicationBaseView {
   }
 
   public Content render(
-      Request request, String programTitle, Optional<String> applicantName, Messages messages) {
+      Request request,
+      String programTitle,
+      Optional<String> applicantName,
+      Messages messages,
+      ImmutableList<String> questions,
+      String redirectToApply,
+      String redirectToEdit,
+      String redirectToProgramDetails) {
```

Comment:
**suggestion (blocking):** This is quite a few parameters, most of them primitives.

Recommend passing a `ReadOnlyApplicantProgramService` and adding methods to it for accessing the applicant and program. With that you'll be able to do derive the values for most of the new parameters here in this class rather than in the code that consumes it.

Reply:
Makes sense! I also passed applicantId and programId instead of passing the URLs (to retrieve them in the view). Let me know what you think.